api/models/userDao.js

```diff
@@ -0,0 +1,40 @@
+const appDataSource = require('./dataSource')
+
+const signUp = async(kakaoUserInfo, res) => {
+    const signUpUser = await appDataSource.query(
+        `INSERT INTO 
+            users (
+                name,
+                email,
+                gender,
+                social_id
+            ) VALUES (
+                ?,
+                ?,
+                ?,
+                ?
+            )
+            `, [kakaoUserInfo.nickname, 
+                kakaoUserInfo.email, 
+                kakaoUserInfo.gender, 
+                kakaoUserInfo.sub]
+        )
+        return signUpUser
+}
+
+const getUserSocialId = async(socialId) => {
+    const userInfo = await appDataSource.query(
+        `SELECT
+            u.id,
+            u.name,
+            u.email,
+            u.gender,
+            u.social_id
+        FROM users u
+        WHERE u.social_id = ?
+        `, [socialId]
+    )
+    return userInfo[0]
+}
```

Comment:
getUserSocialId라는건 유저의 소셜아이디를 get하겠다는 것 처럼 보일 수 있어서 조금 더 명확히 보이면 좋을 것 같아요-!

또한 어차피 query를 통해 return 되는 data가 특정 object 1개인데 배열로 감싸져 있는거라면 다음과 같이 작성하는 것이 코드 가독성에서 조금 더 나을 것 같습니다.
```suggestion
const getUserBySocialId = async(socialId) => {
    const [userInfo] = await appDataSource.query(
        `SELECT
            u.id,
            u.name,
            u.email,
            u.gender,
            u.social_id
        FROM users u
        WHERE u.social_id = ?
        `, [socialId]
    )
    return userInfo
}
```